components/raftstore/src/store/peer.rs

```diff
@@ -3722,12 +3722,19 @@ where
         ctx.raft_metrics.propose.transfer_leader += 1;
 
         let transfer_leader = get_transfer_leader_cmd(&req).unwrap();
-        let peer = transfer_leader.get_peer();
+        let prs = self.raft_group.raft.prs();
 
-        let transferred = if peer.id == self.peer.id {
-            false
-        } else {
-            self.pre_transfer_leader(peer)
+        let peer = transfer_leader
+            .get_peers()
+            .iter()
+            .chain([transfer_leader.get_peer()])
+            .max_by_key(|c| prs.get(c.id).map_or(0, |pr| pr.matched));
+
+        let transferred = match peer {
+            Some(peer) if peer.id != self.peer.id && *peer != metapb::Peer::default() => {
```

Comment:
Actually there may exist `peer` equal to `metapb::Peer::default()` especially in test.

Reply:
What is the expected behavior for the default peer, 0.0? Still call `pre_transfer_leader` with it?

Reply:
What I mean is we should not rely on `peer == Peer::default()` to determine whether the `peer` field is set, because the field may exactly be set to `Peer::default()`. Should use `xxx.has_peer()` and `xxx.has_peers()` instead.